proxy-robots-module/src/main/java/orwell/proxy/zmq/ZmqMessageBOM.java

```diff
@@ -29,42 +31,47 @@ public ZmqMessageBOM(final String routingId, final EnumMessageType messageType,
     /**
      * @param raw_zmq_message Message receive through zmq protocol, with
      *                        the following format (split by spaces):
-     *                        routingID typeString message
+     *                        routingId typeString message
+     *                        Careful: message is arbitrary binary data
      */
     public static ZmqMessageBOM parseFrom(final byte[] raw_zmq_message) throws ParseException {
-        final String zmqMessageString = new String(raw_zmq_message);
-        final String[] zmqMessageStringArray = zmqMessageString.split(" ", 3);
+        // We do not want to create a String from arbitrary binary data, so we
+        // first isolate the 3 parts of the raw zmq message
+        final List<byte[]> zmqMessageBytesList = Utils.split(ZMQ_SEPARATOR, raw_zmq_message, 3);
```

Comment:
There is a magic number and it explanation which suggests a dedicated constant would be better.